lib/task-queue/Storage.js

```diff
@@ -0,0 +1,146 @@
+/* globals XPCOMUtils, OS, Sqlite, Task, DB_PATH */
+"use strict";
+
+const {Cu} = require("chrome");
+
+Cu.import("resource://gre/modules/XPCOMUtils.jsm");
+Cu.import("resource://gre/modules/osfile.jsm");
+Cu.import("resource://gre/modules/Task.jsm");
+
+XPCOMUtils.defineLazyModuleGetter(this, "Sqlite",
+                                  "resource://gre/modules/Sqlite.jsm");
+
+XPCOMUtils.defineLazyGetter(this, "DB_PATH", function() {
+  return OS.Path.join(OS.Constants.Path.profileDir, "taskqueue.sqlite");
+});
+
+// SQL for creating the moz_pages table
+const CREATE_PAGES_TABLE = `CREATE TABLE moz_pages (
+  url LONGVARCHAR PRIMARY KEY,
+  path LONGVARCHAR,
+  expiry DATE,
+  createdAt DATE,
+  remote BOOLEAN DEFAULT 0 NOT NULL CHECK (remote IN (0,1))
+);`;
+
+// SQL for droping the moz_pages table
+const DROP_PAGES_TABLE = "DROP TABLE moz_pages;";
+
+// SQL for creating the moz_tasks table
+const CREATE_TASKS_TABLE = `CREATE TABLE moz_tasks (
+  id INTEGER PRIMARY KEY ASC,
+  pageUrl LONGVARCHAR,
+  createdAt INTEGER,
+  jobStartedAt INTEGER,
+  status LONGVARCHAR,
+  type LONGVARCHAR
+);`;
+
+// SQL for droping the moz_tasks table
+const DROP_TASKS_TABLE = "DROP TABLE moz_tasks;";
+
+class Storage {
+  /**
+   * Handles database storage needs for the task queue
+   */
+  constructor() {
+    this._conn = null;
+  }
+
+  /**
+   * Creates the database tables for the job infrastructure
+   */
+  createTables() {
+    return Task.spawn((function*() {
+      yield this._getConnection().then((db)=>db.execute(CREATE_PAGES_TABLE));
+      yield this._getConnection().then((db)=>db.execute(CREATE_TASKS_TABLE));
+    }).bind(this));
+  }
+
+  /**
+   * Removes the database tables used for the job infrastructure
+   */
+  dropTables() {
+    return Task.spawn((function*() {
+      yield this._getConnection().then((db)=>db.execute(DROP_PAGES_TABLE));
+      yield this._getConnection().then((db)=>db.execute(DROP_TASKS_TABLE));
+    }).bind(this));
+  }
+
+  /**
+   * Close the connection to the database
+   */
+  closeConnection() {
```

Comment:
Please make this a `Task.async` function and prefix the function name with `async`